notifier/notifier.go

```diff
@@ -126,14 +101,50 @@ func New(s *storage.Storage, concurrency int, logger *log.Logger, dwnlURL string
 
 // Start starts the Notifier loop and instruments the worker goroutines that
 // perform the actual notify requests.
-func (n *Notifier) Start(closeChan chan struct{}) {
+func (n *Notifier) Start(closeChan chan struct{}, cfg map[string]interface{}) {
+	ctx, cancelfunc := context.WithCancel(context.Background())
+
+	for id, v := range cfg {
+		if len(v.(map[string]interface{})) == 0 {
+			continue
+		}
+
+		if id == HTTP {
+			n.backends[id] = &httpbackend.Backend{}
+		} else if id == KAFKA {
+			n.backends[id] = &kafkabackend.Backend{}
+		}
+
+		b := n.backends[id]
+		n.Log.Printf("\nStarting %s backend", b.ID())
+		err := b.Start(ctx, v.(map[string]interface{}))
+		if err != nil {
+			n.Log.Printf("Error while initializing backend %s. Error details: %s", b.ID(), err)
+		}
+	}
+
+	if len(n.backends) == 0 {
```

Comment:
I think we should exit if no backends are available.

Reply:
..with a non-zero error code